org/qmckl_sherman_morrison_woodbury.org

```diff
@@ -0,0 +1,1775 @@
+    - ~later_updates~ is allocated with at least $1 \times 2 \times 8$ bytes
+    - ~later_index~ is allocated with at least $1 \times 8$ bytes
+    - ~later >= 0~ 
+      
+*** C header
+
+    #+CALL: generate_c_header(table=qmckl_slagel_splitting_args,rettyp=get_value("CRetType"),fname=get_value("Name"))
+
+    #+RESULTS:
+    #+begin_src c :tangle (eval h_func) :comments org
+    qmckl_exit_code qmckl_slagel_splitting_c (
+	  const uint64_t Dim,
+	  const uint64_t N_updates,
+	  const double* Updates,
+	  const uint64_t* Updates_index,
+	  const double breakdown,
+	  double* Slater_inv,
+	  double* later_updates,
+	  uint64_t* later_index,
+	  uint64_t* later ); 
+    #+end_src
+
+*** Source Fortran
+
+*** Source C
+
+    #+begin_src c :tangle (eval c) :comments org
+#include <stdbool.h>
+#include <math.h>
+#include "qmckl.h"
+
+qmckl_exit_code qmckl_slagel_splitting_c(uint64_t Dim,
+                                              uint64_t N_updates,
+                                              const double *Updates, 
+                                              const uint64_t *Updates_index,
+                                              const double breakdown,
+                                              double *Slater_inv,
+                                              double *later_updates, 
+                                              uint64_t *later_index,
+                                              uint64_t *later) {
+// #ifdef DEBUG //  Leave commented out since debugging information is not yet implemented in QMCkl.
+//   std::cerr << "Called slagel_splitting with " << N_updates << " updates" << std::endl;
+// #endif
+
+  double C[Dim];
+  double D[Dim];
+
+  uint64_t l = 0;
+  // For each update
+  while (l < N_updates) {
+    // C = S^{-1} x U_l
+    for (uint64_t i = 0; i < Dim; i++) {
+      C[i] = 0;
+      for (uint64_t j = 0; j < Dim; j++) {
+        C[i] += Slater_inv[i * Dim + j] * Updates[l * Dim + j];
+      }
+    }
+
+    // Denominator
+    double den = 1 + C[Updates_index[l] - 1];
+    if (fabs(den) < breakdown) {
+
+      // U_l = U_l / 2 (do the split)
+      for (uint64_t i = 0; i < Dim; i++) {
+        later_updates[*later * Dim + i] = Updates[l * Dim + i] / 2.0;
+        C[i] /= 2.0;
+      }
+      later_index[*later] = Updates_index[l];
+      (*later)++;
+
+      den = 1 + C[Updates_index[l] - 1];
+    }
+    double iden = 1 / den;
+
+    // D = v^T x S^{-1}
+    for (uint64_t j = 0; j < Dim; j++) {
+      D[j] = Slater_inv[(Updates_index[l] - 1) * Dim + j];
+    }
+
+    // S^{-1} = S^{-1} - C x D / den
+    for (uint64_t i = 0; i < Dim; i++) {
+      for (uint64_t j = 0; j < Dim; j++) {
+        double update = C[i] * D[j] * iden;
+        Slater_inv[i * Dim + j] -= update;
+      }
+    }
+    l += 1;
+  }
+
+  return QMCKL_SUCCESS;
+}
+
+    #+end_src
+    
+*** Performance
+
+This function performce better for cycles with 1 rank-1 update and with a high fail-rate.
```

Comment:
Check and eventually update the performance recommendations following new report.